R/ash.R

```diff
@@ -248,15 +248,19 @@ ash.workhorse <-
   # do not put big weight on null component
   # automatically estimate the mode if not specified
   if(sum(lik$name=="pois")){
```

Comment:
If I remove the sum, then when I run normal ash with non-zero mode, it gives the following error message...I'm not sure what's going on here...

> ash.norm.out = ash(rnorm(100,1,2), 2, mode="estimate", prior="uniform")
 Hide Traceback
 
 Rerun with Debug
 Error in if (lik$name == "pois") { : argument is of length zero 
4 ash.workhorse(betahat, sebetahat, mixcompdist = mixcompdist, 
    df = df, ...) 
3 stopifnot(is.list(x), is.list(val)) 
2 utils::modifyList(ash.workhorse(betahat, sebetahat, mixcompdist = mixcompdist, 
    df = df, ...), list(call = match.call())) at ash.R#74
1 ash(rnorm(100, 1, 2), 2, mode = "estimate", prior = "uniform")